src/Gui/QuantitySpinBox.cpp

```diff
@@ -585,18 +585,19 @@ void QuantitySpinBox::openFormulaDialog()
     Q_EMIT showFormulaDialog(true);
 }
 
-void QuantitySpinBox::handlePendingEmit()
+void QuantitySpinBox::handlePendingEmit(bool updateUnit /* = true */)
 {
-    updateFromCache(true);
+    updateFromCache(true, updateUnit);
 }
 
-void QuantitySpinBox::updateFromCache(bool notify)
+void QuantitySpinBox::updateFromCache(bool notify, bool updateUnit /* = true */)
 {
     Q_D(QuantitySpinBox);
     if (d->pendingEmit) {
         double factor;
         const Base::Quantity& res = d->cached;
-        QString text = getUserString(res, factor, d->unitStr);
+        auto tmpUnit(d->unitStr);
+        QString text = getUserString(res, factor, updateUnit?d->unitStr:tmpUnit);
         d->unitValue = res.getValue() / factor;
```

Comment:
This might be a problem because `d->unitValue` should be in sync with `d->unitStr`.